Turning off globe turns off defaults for baseLayerPicker & imageryProvider

emackey opened the pull request:
This is to address an idea @mramato had in #2535.

pjcozzi commented:
@mramato merge when you are happy.

mramato commented on Source/Widgets/CesiumWidget/CesiumWidget.js:
```diff
@@ -192,8 +192,9 @@ define([
-        if (defined(options) && defined(options.globe) && options.globe === false && options.imageryProvider !== false) {
-            throw new DeveloperError('To turn off the globe, options.imageryProvider must be set to false.');
+        if (defined(options) && defined(options.globe) && options.globe === false &&
+                defined(options.imageryProvider) && options.imageryProvider !== false) {
```
I'm not crazy about the working on the error but I would prefer that we don't even do this check, if globe is off just silently ignore the other stuff.  Otherwise there are a host of other options that need to be checked: `selectedImageryProviderViewModel`, `imageryProviderViewModels`, `selectedTerrainProviderViewModel`, `terrainProviderViewModels`, and `terrainProvider`, possible more.

mramato commented on Source/Widgets/CesiumWidget/CesiumWidget.js:
```diff
@@ -294,7 +295,7 @@ define([
 
             //Set the base imagery layer
-            var imageryProvider = options.imageryProvider;
+            var imageryProvider = (options.globe === false) ? false : options.imageryProvider;
```
Similar checks probably need to be added for the properties I mentioned in my previous comment.

mramato commented:
Just those 2 comments.

emackey commented:
I removed the error messages and added the check for `terrainProvider`.  No checks are needed for `selectedImageryProviderViewModel`, `imageryProviderViewModels`, and such because they are passed into the baseLayerPicker, which is turned off.

emackey commented:
@mramato I think this is ready.
